### PR TITLE
Expose secure-boot-enroll systemd-boot option

### PR DESCRIPTION
**Pull request by jimmykarily**
Fixes https://github.com/kairos-io/kairos/issues/2430

### CODE REVIEW

**Comment by jimmykarily** [comment]
We can document this here: https://github.com/kairos-io/kairos-docs/pull/168/files

File: cmd/build-uki.go [hunk 1]
```diff
@@ -150,6 +150,8 @@ func NewBuildUKICmd() *cobra.Command {
 	c.Flags().StringP("keys", "k", "", "Directory with the signing keys")
 	c.Flags().StringP("default-entry", "e", "", "Default entry selected in the boot menu.\nSupported glob wildcard patterns are \"?\", \"*\", and \"[...]\".\nIf not selected, the default entry with install-mode is selected.")
 	c.Flags().Int64P("efi-size-warn", "", 1024, "EFI file size warning threshold in megabytes. Default is 1024.")
+	c.Flags().String("secure-boot-enroll", "if-safe", "The value of secure-boot-enroll option of systemd-boot. Possible values: off|manual|if-safe|force. Minimum systemd version: 253. Docs: https://manpages.debian.org/experimental/systemd-boot/loader.conf.5.en.html. !! Danger: this feature might soft-brick your device if used improperly !!")
```
Comment (mauromorales) [inline cmd/build-uki.go:153]: oh wow, scary 👻

File: e2e/build_uki_test.go [hunk 2]
```diff
@@ -56,19 +52,49 @@ var _ = Describe("build-uki", func() {
...
-		fmt.Println(out)
-
-		out, err = enki.Run("build-uki", image, "--output-dir", resultDir, "-k", keysDir, "--output-type", "iso")
-		Expect(err).ToNot(HaveOccurred(), out)
-
-		By("building the iso")
-		_, err = os.Stat(resultFile)
-		Expect(err).ToNot(HaveOccurred())
+	Describe("building an iso", func() {
+		When("secure-boot-enroll is not set", func() {
```
Comment (mauromorales) [inline e2e/build_uki_test.go:56]: IMO having these `When` sections but not putting a differenciation between the other `When` brings little value. We could just have an `It("successfully builds a uki iso from a container image when secure-boot-enrool is not set`

So for a more Ginkgo approach I think we should use a JustBeforeEach here to set those parameters

File: cmd/build-uki.go (hunk 1 above)
Comment (jimmykarily) [reply cmd/build-uki.go:153]: yup, that was the purpose :D

File: pkg/action/build-uki.go [hunk 3]
```diff
@@ -182,8 +182,14 @@ func (b *BuildUKIAction) createSystemdConf(sourceDir string) error {
 		// This is the one name that has nothing added, just the version
 		finalEfiConf = nameFromCmdline(constants.UkiCmdline+" "+constants.UkiCmdlineInstall) + ".conf"
 	}
+
+	secureBootEnroll := viper.GetString("secure-boot-enroll")
+	if secureBootEnroll == "" {
```
Comment (Itxaka) [inline pkg/action/build-uki.go:187]: no need to do this if the flags are binded no? I thought we fixed this?

Comment (jimmykarily) [reply pkg/action/build-uki.go:187]: You are right. Removing.

**Pull request merged**